ilastik/applets/labeling/labelExplorer.py

```diff
@@ -0,0 +1,271 @@
+        through `nonzero_block_slot`. -> Neighbourhood.SINGLE
+        Note: Neighbourhood.NDIM would lead to perfect results in every situation. In practice considering
+        Neighbourhood,SINGLE would diverge from the perfect result only if the label has exactly 1Pixel wide
+        annotations in diagonal blocks.
+      * if `label_slot.meta.is_blocked_cache` is not True it indicates a non-blocked cache - as used in
+        Carving and Counting. In this case it is assumed that any update will require update of the
+        whole label array. -> Neighbourhood.NONE
+
+    The ui is lazy. Calculations are only done if this widget is shown. If changes to the
+    label slot are detected without the ui being visible, `_table_initialized` is set to
+    False and a complete recalculation will happen the next time the widget is shown.
+    (This could be in principle optimized by keeping track of the dirty rois.)
+    """
+
+    positionRequested = Signal(dict)
+
+    display_text = "Label Explorer"
+
+    def __init__(
+        self,
+        nonzero_blocks_slot: OutputSlot,
+        label_slot: OutputSlot,
+        parent=None,
+    ):
+        super().__init__(parent)
+        self._table_initialized: bool = False
+        self._shown: bool = False
+        self._lookup_table: dict[str, str] = {}
+        self._nonzero_blocks_slot = nonzero_blocks_slot
+        self._axistags = label_slot.meta.getAxisKeys()
+        self._display_axistags = [x for x in self._axistags if x != "c"]
+        self._item_flags = Qt.ItemIsSelectable | Qt.ItemIsEnabled | Qt.ItemNeverHasChildren
+        self._label_slot = label_slot
+        self._block_cache: Dict[Tuple[int, ...], Block] = {}
+        self._neighbourhood = (
+            Neighbourhood.SINGLE if getattr(self._label_slot.meta, "is_blocked_cache", False) else Neighbourhood.NONE
+        )
+
+        self._setupUi()
+
+        self.unsubscribe_fns = []
+        self.unsubscribe_fns.append(label_slot.notifyDirty(self.update_table))
+
+        def _sync_viewer_position(currentRow, _currentColumn, _previousRow, _previousColumn):
+            position = self.tableWidget.item(currentRow, 0).data(Qt.UserRole)
+            self.positionRequested.emit(position)
+
+        self.tableWidget.currentCellChanged.connect(_sync_viewer_position)
+
+    def _clear_blocking(self):
+        self._block_cache = {}
+
+    def _item_lookup(self, item):
+        return self._lookup_table.get(item, "NOT FOUND")
+
+    def paintEvent(self, a0: QPaintEvent) -> None:
+        # Hack: Initialize new widget after lane removal
+        # need to trigger initialization manually - while the widget is on the
+        # not on the screen
+        if not self._table_initialized:
+            QTimer.singleShot(0, self.sync_state)
+        super().paintEvent(a0)
+
+    def _setupUi(self):
+        layout = QVBoxLayout()
+        self.tableWidget = QTableWidget()
```

Comment:
```suggestion
    def _setupUi(self):
        layout = QVBoxLayout()
        layout.setContentsMargins(0, 0, 0, 0)
        self.tableWidget = QTableWidget()
```
Found the source of the irritating empty space